kong/plugins/response-ratelimiting/policies/init.lua

```diff
@@ -208,8 +208,13 @@ return {
     usage = function(conf, identifier, name, period, current_timestamp)
       local red = redis:new()
       red:set_timeout(conf.redis_timeout)
-
-      local ok, err = red:connect(conf.redis_host, conf.redis_port)
+      -- use a special pool name only if redis_database is set to non-zero
+      -- otherwise use the default pool name host:port
+      sock_opts.pool = conf.redis_database and
```

Comment:
Related to my above comment (when mentioning "always specifying a pool name): if `redis_database` is `nil`, this call to `connect` may use a previously-set `sock_opts.pool` value (since `sock_opts` is an upvalue and we don't clear the field when `conf.redis_database == nil`). That's a bug.

Reply:
Oh, my bad, it should be fine actually.